### PR TITLE
Add container scanning for OSS images

### PR DESCRIPTION
**Pull request by ciarams87**
### Proposed changes
Add trivy container scanning for OSS images. Scanning of the Plus and App Protect images will be done separately in GitLab.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have rebased my branch onto master

### CODE REVIEW

File: .github/workflows/edge.yml [hunk 1]
```diff
@@ -92,6 +92,64 @@ jobs:
...
+          build-args: |
+            BUILD_OS=${{ matrix.image }}
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: '${{ matrix.image }}:${{ github.sha }}'
+          format: 'template'
+          template: '@/contrib/sarif.tpl'
+          output: 'trivy-results-${{ matrix.image }}.sarif'
+          severity: 'CRITICAL,HIGH'
```
Comment (lucacome) [inline .github/workflows/edge.yml:137]: should we also include `LOW` and `MEDIUM`? I think if we enable `ignore-unfixed: true` it makes sense to have them all.

Comment (lucacome) [inline .github/workflows/edge.yml:137]: we should add `ignore-unfixed: true` since if they're not fixed upstream there's not much we can do.

**Review by lucacome — CHANGES_REQUESTED** [review]
Do we need to build the images again? Can't it just be a step in one of the other builds?

**Comment by ciarams87** [comment]
> Do we need to build the images again? Can't it just be a step in one of the other builds?

It could just be a step in another build, but having it as a separate stage runs these jobs concurrently with the other helm and smoke tests, so it doesn't take any longer, and I think it's easier to follow this way.

**Review by pleshakov — APPROVED** [review]

**Pull request merged**